one_fm/api/doc_methods/stock_entry.py

```diff
@@ -180,4 +180,29 @@ def validate_stock_entry_items(doc, method):
                     obj.basic_amount = 0
                     obj.valuation_rate = 0
                     obj.additional_cost = 0
-                    obj.basic_rate = 0
\ No newline at end of file
+                    obj.basic_rate = 0
+                    
+    elif doc.stock_entry_type == "Material Issue":
+        if doc.items:
+            list_of_items = [i.item_code for i in doc.items]
+            difference_account_field = frappe.db.get_all("Item", {"name": ["in", list_of_items]}, ["name", "difference_account", "subitem_group"])
+            account_check = {item["name"]: {"account": item["difference_account"], "group": item["subitem_group"]} for item in difference_account_field}
+            for obj in doc.items:
+                account = account_check[obj.item_code].get("account", False)
+                group = account_check[obj.item_code].get("group", False)
+                if account:
+                    obj.expense_account = account  
+                else:
+                    if group:
+                        if group == "Uniform":
+                            obj.expense_account = "Stock Issued - ONEFM"

```

Comment:
Hard coding is not the correct approach. The group name can be renamed, the code will not work properly.

Reply:
@pjamsheer Yes, i agree, so i added a field called difference_account to the item doctype, so it fetches from that field, the only time it gets to the hard coded part was when the account was not added to the item.